bolt/lib/Core/BinaryFunction.cpp

```diff
@@ -1961,6 +1961,44 @@ bool BinaryFunction::postProcessIndirectBranches(
       bool IsEpilogue = llvm::any_of(BB, [&](const MCInst &Instr) {
         return BC.MIB->isLeave(Instr) || BC.MIB->isPop(Instr);
       });
+      if (BC.isAArch64()) {
+        // Any ADR instruction of AArch64 will generate a new entry,
+        // ADR instruction cannot afford to do any optimizations. Because ADR
+        // computes a PC-relative address within a limited range tied to the
+        // current program counter, optimizing transformations (like code
+        // rearrangements) can change address distances and potentially exceed
+        // ADR’s range.
+        if (!IsEpilogue && !isMultiEntry()) {
+          BinaryBasicBlock::iterator LastDefCFAOffsetInstIter = BB.end();
+          // Find the last OpDefCfaOffset 0 instruction.
+          for (BinaryBasicBlock::iterator Iter = BB.begin(); Iter != BB.end();
```

Comment:
I'm a bit confused about your goal here. IIUC you need the last CFI before the branch, right?
Wouldn't this loop retrieve the first CFI of the whole block instead?

If that's the case, wouldn't you need to start from `II` and go backward until you hit the first CFI?